### DISCUSSION TITLE
Random questions on reported capacity and volume assignment via Filer(s)

### DISCUSSION THREAD

**Discussion by prologic** [seaweedfs/seaweedfs#2107] [Q&A]
Hi Chris 👋

Just some random questions and observations I have playing with a test SeaweedFS cluster over the last couple of days.

Setup:

- 3x SeaweedFS nodes
  - 1x Master + Filer node
  - 2x Volume nodes

Each node has 3x 30GB disks `/dev/sdb`, `/dev/sdc` and `/dev/sdd` each partitioned and formatted as EXT4 and mounted to `/mnt/vol1`, `/mnt/vol2` and `/mnt/vol3`.

Master is run with:

```#!sh
$ weed server -dir=/mnt/vol1,/mnt/vol2,/mnt/vol3 -filer
```

Each volume node is run with:

```#!sh
$ weed volume -mserver <master_ip>:9333 -dir=/mnt/vol1,/mnt/vol2,/mnt/vol3
```

Some questions and observations:

- When I `weed mount` a filer path on another machine I see the reported capacity as ~206G -- Where does this number come from? We assume this is roughly 7x 30GB?
- Adding more volume nodes doesn't seem to increase reported capacity at all.
- In testing no data appears to be written to or read from other volume nodes. Why? There are no reported errors.

**Comment by chrislusf** [comment]
* allocated_volume_count * volume size = capacity
* ^^
* data writes to allocated volumes. You may need to run a `volume.balance` in `weed shell`.

**Reply by prologic** [reply]
> * allocated_volume_count * volume size = capacity

When does allocation happen? Only when volumes are actually allocated to a new volume server?

> * data writes to allocated volumes. You may need to run a `volume.balance` in `weed shell`.

Similar to above question; how are volume servers and their volumes picked/allocated?

In my observation I see basically all volumes allocated on the master volume server only. Is it expected to run `volume.balance` periodically? Is there some strategy in the master for picking and allocating volumes to volume servers -- or is the first one always picked? 🤔

----

Also as an aside (_which I'm hoping is something I've done wrong in my setup_) I'm seeing some worrying I/O errors when reading files. I ran `fio` a few times with various file sizes 1M 10M 100M and 1G and with 100M and 1G files I cannot compute an `md5sum` of them. I get read i/o errors.

**Reply by chrislusf** [reply]
volumes are created when existing volumes almost runs out.
they are created randomly weighted by volume server emptiness.

it is part of admin scripts, which by default will run every 17 minutes.

need to make it a reproducible bug.
